### PR TITLE
Login module

### PR DESCRIPTION
**Pull request by urbanek32**

### CODE REVIEW

File: front-end/app/scripts/app.js [hunk 1]
```diff
@@ -7,10 +7,16 @@
...
     angular.module('openvillage', [
             'ui.router',                    // Routing
-            'ui.bootstrap'                 // Bootstrap
+            'ui.bootstrap',                 // Bootstrap
+            'ui.router.login'
         ])
-        .config(function ($locationProvider) {
-            $locationProvider.html5Mode(true);
-        });
+        /*.config(function ($locationProvider) {
```
Comment (ajoskows) [inline front-end/app/scripts/app.js]: If it is unnecessary, remove it please

File: front-end/app/scripts/config.js [hunk 2]
```diff
@@ -8,60 +8,99 @@
...
             abstract: true,
             url: '/index',
-            templateUrl: 'views/common/content.html',
+            templateUrl: 'views/common/content.html'
         })
         .state('index.main', {
             url: '/main',
             templateUrl: 'views/main.html',
-            data: { pageTitle: 'Example view' }
+            //controller: 'UsersCtrl',
```
Comment (ajoskows) [inline front-end/app/scripts/config.js]: remove it

File: front-end/app/scripts/config.js [hunk 3]
```diff
@@ -8,60 +8,99 @@
...
+        .setAuthClearMethod('clearAuthKey')
+        .setAuthGetMethod('getAuthKey')
+        .setCookieName('__loginState');
+
 }
 angular
     .module('openvillage')
     .config(config)
     .run(function($rootScope, $state) {
         $rootScope.$state = $state;
```
Comment (ajoskows) [inline front-end/app/scripts/config.js]: Remove unnecessary lines

File: front-end/app/scripts/controllers.js [hunk 4]
```diff
@@ -8,13 +8,23 @@
  */
 'use strict';
 
-function MainCtrl($window) {
+function MainCtrl($scope, $window, projectsService) {
```
Comment (ajoskows) [inline front-end/app/scripts/controllers.js]: Remove unnecessary line

File: front-end/app/scripts/controllers.js [hunk 5]
```diff
@@ -8,13 +8,23 @@
...
     this.userName = 'Example user';
     this.helloText = 'Welcome in SeedProject';
     this.descriptionText = 'It is an application skeleton for a typical AngularJS web app. You can use it to quickly bootstrap your angular webapp projects and dev environment for these projects.';
 
     this.userName = $window.sessionStorage.sessionUsername;
+
+    $scope.numberOfProjectsOwned = -1;
+    projectsService.getList()
+        .then(function (result) {
+            //console.log(result.length);
```
Comment (ajoskows) [inline front-end/app/scripts/controllers.js]: Remove

File: front-end/app/scripts/controllers.js [hunk 6]
```diff
@@ -8,13 +8,23 @@
...
+            $scope.numberOfProjectsOwned = result.length;
+        }, function (err) {
+            //$scope.numberOfProjectsOwned = err;
```
Comment (ajoskows) [inline front-end/app/scripts/controllers.js]: remove all console logs and commented out lines

File: front-end/app/scripts/services/UserAuth.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+angular
+    .module('openvillage')
+    .service('userAuthService', function ($rootScope, $q, $window) {
+
+        this.getAuthKey = function () {
+
+            //console.log('get auth');
```
Comment (ajoskows) [inline front-end/app/scripts/services/UserAuth.js]: remove

File: front-end/app/scripts/services/UserAuth.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+            if($window.sessionStorage.token && $window.sessionStorage.sessionUsername) {
+                return true;
+            }
+
```
Comment (ajoskows) [inline front-end/app/scripts/services/UserAuth.js]: remove this line

File: front-end/app/scripts/services/UserAuth.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+            return false;
+        };
+
+        this.clearAuthKey = function () {
+
```
Comment (ajoskows) [inline front-end/app/scripts/services/UserAuth.js]: remove these two lines

File: front-end/app/views/common/navigation.html [hunk 10]
```diff
@@ -13,7 +13,9 @@
                             </span>
                     </a>
                     <ul uib-dropdown-menu class="animated flipInX m-t-xs">
-                        <li><a href="">Logout</a></li>
+                        <li><a ui-sref="index.main">Main</a></li>
```
Comment (ajoskows) [inline front-end/app/views/common/navigation.html]: What dou you thing about writing some code using 'ng-repeat' function? You can define menu items in controller and later replace this code using three lines. Adding new menu items will be more comfortable

File: front-end/app/views/main.html [hunk 11]
```diff
@@ -1,5 +1,63 @@
...
+                </div>
+                <div class="ibox-content">
+                    <h1 class="no-margins">{{numberOfProjectsOwned}}</h1>
+
+                    <!--<div class="stat-percent font-bold text-success">98% <i class="fa fa-bolt"></i></div>-->
+                    <small>Number of projects</small>
+                </div>
+            </div>
+        </div>
+        <!--<div class="col-lg-3">
```
Comment (ajoskows) [inline front-end/app/views/main.html]: remove all commented out code

**Comment by ajoskows** [comment]
It works and looks almost OK. Check my comments and fix the code, later you can merge your changes ;)

**Comment by ajoskows** [comment]
Before merging squash commits, because second commit looks like temporary commit

File: front-end/app/views/common/navigation.html (hunk 10 above)
Comment (urbanek32) [inline front-end/app/views/common/navigation.html]: Will think about it later, now its useless.

**Pull request merged**